server/src/main/java/io/druid/segment/realtime/firehose/EventReceiverFirehoseFactory.java

```diff
@@ -146,12 +183,18 @@ public Response addAll(Collection<Map<String, Object>> events)
           }
         }
 
-        return Response.ok().entity(ImmutableMap.of("eventCount", events.size())).build();
+        return Response.ok(
+            objectMapper.writeValueAsString(ImmutableMap.of("eventCount", events.size())),
+            contentType
```

Comment:
the general contract that the existing apis follow is that if the content received is smile, the response is also sent in smile format. e.g. in QueryResource. Its better to set content type if someone relies on that to parse the response.